src/lib/zkapp.ts

```diff
@@ -555,24 +571,28 @@ class SmartContract {
     });
   }
 
-  static async compile(address: PublicKey, tokenId: Field = TokenId.default) {
+  static async compile() {
```

Comment:
While we're changing `compile` and `analyzeMethods` in this PR, could you add doc comments as well if you don't mind? :D

Reply:
nice catch!